security/pkg/pki/ca/ca.go

```diff
@@ -275,15 +275,21 @@ type IstioCA struct {
 // NewIstioCA returns a new IstioCA instance.
 func NewIstioCA(opts *IstioCAOptions) (*IstioCA, error) {
 	ca := &IstioCA{
-		defaultCertTTL: opts.DefaultCertTTL,
-		maxCertTTL:     opts.MaxCertTTL,
-		keyCertBundle:  opts.KeyCertBundle,
-		livenessProbe:  probe.NewProbe(),
+		maxCertTTL:    opts.MaxCertTTL,
+		keyCertBundle: opts.KeyCertBundle,
+		livenessProbe: probe.NewProbe(),
 	}
 
 	if opts.CAType == selfSignedCA && opts.RotatorConfig.CheckInterval > time.Duration(0) {
 		ca.rootCertRotator = NewSelfSignedCARootCertRotator(opts.RotatorConfig, ca)
 	}
+
+	certTTL, err := util.TimeBeforeNextRotate(ca.keyCertBundle.GetCertChainPem(), ca.defaultCertTTL, time.Now())
```

Comment:
you are passing `ca.defaultCertTTL` but it's not set in line 277, shouldn't you pass `opts.DefaultCertTTL`?